File: JavaToKotlin/apps(kotlin)/src/main/kotlin/io/dustin/apps/board/domain/user/repository/UserRepository.java
package io.dustin.apps.board.domain.user.repository;

import io.dustin.apps.board.domain.user.repository.custom.CustomUserRepository;
import io.dustin.apps.common.repository.BaseRepository;
import io.dustin.apps.board.domain.user.model.SiteUser;

import java.util.Optional;

public interface UserRepository extends BaseRepository<SiteUser, Long>, CustomUserRepository {
    Optional<SiteUser> findByNickName(String username);

    Optional<SiteUser> findById(Long userId);

}

File: dustin/src/main/kotlin/io/dustin/domain/user/repository/UserRepository.kt
package io.dustin.domain.user.repository

import io.dustin.domain.user.model.User
import io.dustin.domain.user.repository.custom.CustomUserRepository
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface UserRepository: ReactiveCrudRepository<User, Long>, CustomUserRepository {
    override fun findById(id: Long): Mono<User>
    fun findAllBy(pageable: Pageable): Flux<User>
}